repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,669 @@
+                words = words[2:]
+            elif len(words) >= 2:
+                words = words[2:]
+            else:
+                return False
+
+        self.fudges[address] = options
+        return True
+
+    def parse_restrict(self, words):
+        ipv4_only = False
+        ipv6_only = False
+        flags = set()
+        mask = ""
+
+        if words[0] == "-4":
+            ipv4_only = True
+            words.pop(0)
+        elif words[0] == "-6":
+            ipv6_only = True
+            words.pop(0)
+
+        if not words:
+            return False
+
+        address = words.pop(0)
+
+        while words:
+            if len(words) >= 2 and words[0] == "mask":
+                mask = words[1]
+                words = words[2:]
+            else:
+                if words[0] not in ["kod", "nomodify", "notrap", "nopeer", "noquery",
+                                    "limited", "ignore", "noserve"]:
+                    return False
+                flags.add(words[0])
+                words.pop(0)
+
+        # Convert to IP network(s), ignoring restrictions with hostnames
+        networks = []
+        if address == "default" and not mask:
+            if not ipv6_only:
+                networks.append(ipaddress.ip_network(u"0.0.0.0/0"))
+            if not ipv4_only:
+                networks.append(ipaddress.ip_network(u"::/0"))
+        else:
+            try:
+                if mask:
+                    networks.append(ipaddress.ip_network(u"{}/{}".format(address, mask)))
+                else:
+                    networks.append(ipaddress.ip_network(address))
+            except ValueError:
+                return False
+
+            if (ipv4_only and networks[-1].version != 4) or \
+                    (ipv6_only and networks[-1].version != 6):
+                return False
+
+        for network in networks:
+            self.restrictions[network] = flags
+
+        return True
+
+    def parse_tos(self, words):
+        options = {}
+        while words:
+            if len(words) >= 2 and words[0] in ["minsane", "orphan"]:
+                if not words[1].isdigit():
+                    return False
+                options[words[0]] = int(words[1])
+                words = words[2:]
+            elif len(words) >= 2 and words[0] in ["maxdist"]:
+                # Check if it is a float value
+                if not words[1].replace('.', '', 1).isdigit():
+                    return False
+                options[words[0]] = float(words[1])
+                words = words[2:]
+            else:
+                return False
+
+        self.tos_options.update(options)
+
+        return True
+
+    def parse_includefile(self, words):
+        path = self.root_dir + words[0]
+        if not os.path.isfile(path):
+            return False
+
+        self.parse_ntp_conf(path)
+        return True
+
+    def parse_keys(self, words):
+        keyfile = words[0]
+        path = self.root_dir + keyfile
```

Comment:
Please use os.path.join for all path building operations

Reply:
Fixed.